go/vt/tabletserver/txlogz.go

```diff
@@ -50,22 +51,58 @@ func init() {
 	http.HandleFunc("/txlogz", txlogzHandler)
 }
 
+func ensureInRange(val int, lower int, upper int) int {
+	if val < lower {
+		return lower
+	} else if val > upper {
+		return upper
+	}
+	return val
+}
+
+func parseReqParam(req *http.Request) (time.Duration, int) {
+	timeout := 10
+	limit := 300
+	if ts, ok := req.URL.Query()["timeout"]; ok {
+		if t, err := strconv.Atoi(ts[0]); err == nil && t >= 0 {
+			timeout = t
```

Comment:
I give them an init value because current web page by default won't set timeout and limit param, we simply call /txlogz/. Thus they both will be 0 if no default values are given, which leads to an empty page.